### DISCUSSION TITLE
Transactions - client database_connection.dart error

### DISCUSSION THREAD

**Discussion by MrFahrenheit-55** [serverpod/serverpod#2152] [Q&A] [answered]
Good day

I am trying to use a transaction to update the db for multiple entries. I have followed the documentation for the transactions and added the transaction to the db queries. When running serverpod_generate, the client.dart file gives an error: 

```dart

import 'package:serverpod/src/database/database_connection.dart' as _i7;

```

this package does not exist. Can you only use transactions for models that are related?

An extension of this question: 
can you pass the transaction as an argument to another function. For example

```dart
someFunction(Session session, int id, Transaction transaction) {
// do some stuff in this function
}

await session.dbNext.transaction((transaction) {

final response = someFunction(session, 1, transaction)
return true
});
```

**Comment by SandPod** [comment, marked as answer]
Hey!

I think the issue is that you are trying to use a transaction as a parameter to an endpoint. The fact that `client.dart` file is giving you errors about the import makes me assume it has been picked up when analyzing your endpoints.

Is your method defined in a class that extends `Endpoint`? If so, does the issue disappear if you move that method to a scope that does not extend `Endpoint`?

Transactions are only used when communicating with the database, so they should traditionally be a server-only concept. Therefore `'package:serverpod/src/database/database_connection.dart'` can't be imported to the client.

There should be no limitations to your possibility of passing transactions to other methods.

**Reply by MrFahrenheit-55** [reply]
Hi @SandPod, yes, this is exactly the problem. I have an endpoint with some functions that pass the Transaction as a parameter. 

This is fixed if the method is removed and everything happens in the endpoint.
